lib/order.rb

```diff
@@ -0,0 +1,87 @@
+class Order
+  attr_reader :id, :products, :customer, :fulfillment_status
+
+  @@tax_rate = 0.075
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+
+    unless fulfillment_status == :pending ||
+           fulfillment_status == :paid ||
+           fulfillment_status == :processing ||
+           fulfillment_status == :shipped ||
+           fulfillment_status == :complete
+      raise ArgumentError, "invalid fulfillment status"
+    end
+  end
+
+  def self.products_hash(products_string)
+    items_with_prices = products_string.split(";")
+
+    products_hash = {}
+
+    items_with_prices.each do |item_with_price|
+      array = item_with_price.split(":")
+
+      item = array[0]
+      price = array[1].to_f
+
+      products_hash[item] = price
+    end
+
+    return products_hash
+  end
+
+  def self.all
+    orders_array = []
+
+    CSV.open("data/orders.csv", "r").each do |line|
+      id = line[0].to_i
+      products_string = line[1]
+      products = products_hash(products_string)
+      customer_id = line[2].to_i
+      customer = Customer.find(customer_id)
+      fulfillment_status = line[3].to_sym
+      orders_array << Order.new(id, products, customer, fulfillment_status)
+    end
+
+    return orders_array
+  end
+
+  def self.find(id)
+    Order.all.each { |order| return order if order.id == id }
```

Comment:
Nice way to condense this into one line so it's a lot cleaner. In case you wanted it even SHORTER, you might want to check out: https://ruby-doc.org/core-2.6.1/Enumerable.html#method-i-find